File: commonlib/src/main/java/com/songdehuai/commonlib/net/ResultCallBack.java
package com.songdehuai.commonlib.net;


import com.alibaba.fastjson.JSONObject;
import com.lzy.okgo.callback.AbsCallback;

import java.lang.reflect.ParameterizedType;
import java.lang.reflect.Type;

import okhttp3.Response;

public abstract class ResultCallBack<T> extends AbsCallback<T> {

    @Override
    public T convertResponse(Response response) throws Throwable {
        T data = null;
        Type rootType = getClass().getGenericSuperclass();
        Type type = ((ParameterizedType) rootType).getActualTypeArguments()[0];
        data = JSONObject.parseObject(response.body().string(), type);
        return data;
    }

}

File: commonlib/src/main/java/com/songdehuai/commonlib/net/ResultCallBack.kt
package com.songdehuai.commonlib.net


import android.os.Handler
import android.os.Looper
import com.alibaba.fastjson.JSONObject

import java.lang.reflect.ParameterizedType

import okgo.callback.AbsCallback
import okhttp3.Response

abstract class ResultCallBack<T>() : AbsCallback<T>() {

    @Throws(Throwable::class)
    override fun convertResponse(response: Response): T? {
        var data: T? = null
        val rootType = javaClass.genericSuperclass
        val type = (rootType as ParameterizedType).actualTypeArguments[0]
        data = JSONObject.parseObject<T>(response.body()!!.string(), type)
        return data
    }

}
